Pearson-Advance/frontend-app-skillable
# PADV-1354 frontend - Create table - student Lab summary

## Tickets

- https://agile-jira.pearson.com/browse/PADV-1354

## Description
This PR adds a feature to show the Lab Summary for a clicked Student

## Changes Made

- [x] Centralize Datatable component to reused
- [x] Add loading component for such Datatables 
- [x] Creates files structure to ease the component navigation
- [x] Add event handler when clicking on a Student's name in Roster Table
- [x] Fetch information from skillable-plugin /labinstancesearch endpoint for the given user
- [x] Render's lab summary fetched
- [x] Apply calculations and styles for table's columns as per Figma's request
- [x] Add PropTypes asserting to accomplish linting verification 
- [x] Add breadcumb for navigation

## How To Test

Please have your devstack environment running, run `npm start` and the MFE shall be running in port 2002 then, please go to the following URL: `http://localhost:2002/skillable_plugin/course-tab/courses/[COURSE_ID]/training_labs` where you have to set the `COURSE_ID` in the URL with a valid course id of your devstack installation. EG: `course-v1:demo+demo1+2020` now it would show a table with the users enrolled to the course and a clickable link in the Student's names.

![image](https://github.com/user-attachments/assets/8c3d1031-c0b1-475f-a7b2-41b37eea0881)

Click on a user and retrieve their information, while the request is made it would show a loading animation

![image](https://github.com/user-attachments/assets/03d01cb9-23f5-4c5e-a95f-a31d3f810f78)

If the request is succesfully completed and there are laboratories to show, it shall display this table with the student's lab instances information 

![image](https://github.com/user-attachments/assets/2095e66e-9548-4880-90a8-7b2d3fdeee16)

Loading component shall work every time we access to the table or move between pages, also, as you might notice, there is a breadcrumb that allows the user to go back to the Class Roster Table by just simply clicking "Go back" hyperlink 

## Note

Changes from PR #5 has been removed due to conflicts, they would be place again in further PR.
